src/Controller/Form.php

```diff
@@ -0,0 +1,239 @@
+<?php
+
+/**
+ * Handle Embed Donation Form Route
+ *
+ * @package Give
+ * @since 2.7.0
+ */
+
+namespace Give\Controller;
+
+use Give\Form\LoadTheme;
+use WP_Post;
+use function Give\Helpers\Form\Utils\isProcessingForm;
+use function Give\Helpers\Form\Utils\isViewingForm;
+use function Give\Helpers\Form\Utils\isViewingFormFailedTransactionPage;
+use function Give\Helpers\Form\Utils\isViewingFormReceipt;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Theme class.
+ *
+ * @since 2.7.0
+ */
+class Form {
+
+	/**
+	 * Initialize
+	 *
+	 * @since 2.7.0
+	 */
+	public function init() {
+		add_action( 'template_redirect', array( $this, 'load' ), 0 );
+		add_action( 'init', array( $this, 'loadThemeOnAjaxRequest' ) );
+		add_action( 'init', array( $this, 'embedFormSuccessURIHandler' ), 1, 3 );
+		add_filter( 'give_send_back_to_checkout', array( $this, 'handlePrePaymentProcessingErrorRedirect' ) );
+		// add_filter( 'wp_redirect', array( $this, 'handleOffSiteCheckoutRedirect' ) );
```

Comment:
I commented this code because I was planning to fix this code in next pr but I am removing it from this pr for now and I will recover it later to fix it.
fixed: https://github.com/impress-org/givewp/compare/d591f36912c6...89930a2fb238